JournalController.DeleteJournalItem can't remove deleted journals

## Description of bug
when journal item already soft deleted, and try to hard delete it, null reference exception will throw out because JournalController.DeleteJournalItem only get the undeleted items.

## Error log
2019-04-01 08:55:10.910+08:00 [DESKTOP-VE8R1FL][D:3][T:68][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.NullReferenceException: Object reference not set to an instance of an object.
   at DotNetNuke.Services.Journal.JournalControllerImpl.DeleteJournalItem(Int32 portalId, Int32 currentUserId, Int32 journalId, Boolean softDelete)
   at DotNetNuke.Services.Journal.JournalControllerImpl.DeleteJournalItem(Int32 portalId, Int32 currentUserId, Int32 journalId)

## Affected version
<!-- Check all that apply and add more if necessary -->

* [x] 9.3.0
* [x] 9.2.2
* [x] 9.2.1
* [x] 9.2

--- a/DNN Platform/Library/Services/Journal/JournalControllerImpl.cs
+++ b/DNN Platform/Library/Services/Journal/JournalControllerImpl.cs
@@ -116,7 +116,12 @@ namespace DotNetNuke.Services.Journal
 
         private void DeleteJournalItem(int portalId, int currentUserId, int journalId, bool softDelete)
         {
-            var ji = GetJournalItem(portalId, currentUserId, journalId);
+            var ji = GetJournalItem(portalId, currentUserId, journalId, !softDelete);
+            if (ji == null)
+            {
+                return;
+            }
+
             var groupId = ji.SocialGroupId;
 
             if (softDelete)

